### DISCUSSION TITLE
Integration Tests with jest giving 'cannot log after tests are done'

### DISCUSSION THREAD

**Discussion by ElmarRoberto** [apollographql/apollo-server#7370] [Q&A]
Hi guys,

On my apollo server v4 project, I'm using 'startStandaloneServer' to run it, after that I'm trying to log something, for example:

_index.js_
```js
...

const startApolloServer = async () => {
  const { url } = await startStandaloneServer(...)
  ...
}

startApolloServer.then(() => {
  // try to console
})

...
```

But my integration test is not ending unless I use `--forceExit` in `package.json` command and I'm getting '**cannot log after tests are done, did you forget something?**'. This is all because `startStandaloneServer`. I already tried using `await server.stop`, but it didn't work. Can someone help me, please?

Thanks!

**Comment by trevor-scheer** [comment]
Can you share some more details? A reproduction perhaps? `await server.stop()` is [the approach we take](https://github.com/apollographql/apollo-server/blob/7ff96f5331fbf14c0a25094007f6f05e799ee052/packages/server/src/__tests__/standalone/standaloneSpecific.test.ts#L142) in our integration tests, so that should work.

Is there a promise you're not awaiting in your tests maybe?

**Reply by glasser** [reply]
Your code snippet does not look like a direct copy-and-paste (eg, you don't have parentheses actually calling the `startApolloServer` function) which will make it even harder for us to figure out what's going on.
